codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -0,0 +1,438 @@
+                || location == HttpBinding.Location.PREFIX_HEADERS
+                || location == HttpBinding.Location.LABEL
+                || location == HttpBinding.Location.QUERY;
+    }
+
+    private void generateOperationHttpBindingSerializer(
+            GenerationContext context,
+            OperationShape operation
+    ) {
+        SymbolProvider symbolProvider = context.getSymbolProvider();
+        Model model = context.getModel();
+        GoWriter writer = context.getWriter();
+
+        Shape inputShape = model.getShape(operation.getInput()
+                .orElseThrow(() -> new CodegenException("missing input shape for operation: " + operation.getId())))
+                .orElseThrow(() -> new CodegenException("input shape missing from model"));
+
+        HttpBindingIndex bindingIndex = context.getModel().getKnowledge(HttpBindingIndex.class);
+
+        Map<String, HttpBinding> bindingMap = bindingIndex.getRequestBindings(operation).entrySet().stream()
+                .filter(entry -> isRestBinding(entry.getValue().getLocation()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        List<String> bindingMembers = new ArrayList<>(bindingMap.keySet());
+        bindingMembers.sort(String::compareTo);
+
+        SymbolReference fmtPackageReference = SymbolUtils.createNamespaceReference(GoDependency.FMT);
+
+        SymbolReference restEncoder = getRestEncoderSymbolReference();
+
+        writer.addUseImports(fmtPackageReference);
+        writer.addUseImports(restEncoder);
+
+        Symbol inputSymbol = symbolProvider.toSymbol(inputShape);
+
+        // TODO: This seems awkward for references generated types package?
+        SymbolReference inputSymbolReference = SymbolReference.builder()
+                .symbol(SymbolUtils.createPointableSymbolBuilder("types." + inputSymbol.getName())
+                        .addReference(SymbolUtils.createNamespaceReference(inputSymbol.getNamespace(), "types"))
+                        .build())
+                .build();
+
+        writer.addUseImports(inputSymbolReference);
+
+        String functionName = ProtocolGenerator.getOperationSerFunctionName(inputSymbol, getProtocolName());
+
+        writer.openBlock("func $L(v $P, encoder $P) error {", "}", functionName, inputSymbolReference, restEncoder,
+                () -> {
+                    writer.openBlock("if v == nil {", "}", () -> {
+                        writer.write("return fmt.Errorf(\"unsupported serialization of nil %T\", v)");
+                    });
+
+                    writer.write("");
+
+                    for (int i = 0; i < bindingMembers.size(); i++) {
+                        HttpBinding binding = bindingMap.get(bindingMembers.get(i));
+                        writeHttpBindingMember(writer, model, symbolProvider, binding);
+                        writer.write("");
+                    }
+
+                    writer.write("return nil");
+                });
```

Comment:
It doesn't *really* matter, but you might want to write out a blank line here. In generally you can be pretty fast and loose with the blank lines since excess blank lines will get stripped away (either by GoWriter or `go fmt`).